edx_proctoring/views.py

```diff
@@ -349,25 +352,36 @@ def get(self, request):
                         data={'detail': _('Must be a Staff User to Perform this request.')}
                     )
 
-        # If there are multiple onboarding exams, use the first exam
-        onboarding_exam = (ProctoredExam.get_practice_proctored_exams_for_course(course_id)
-                           .order_by('-created').first())
-        if not onboarding_exam or not get_backend_provider(name=onboarding_exam.backend).supports_onboarding:
+        # If there are multiple onboarding exams, use the first exam accessible to the user
+        onboarding_exams = list(ProctoredExam.get_practice_proctored_exams_for_course(course_id).order_by('-created'))
+        if not onboarding_exams or not get_backend_provider(name=onboarding_exams[0].backend).supports_onboarding:
             return Response(
                 status=404,
                 data={'detail': _('There is no onboarding exam related to this course id.')}
             )
 
+        learning_sequences_service = get_runtime_service('learning_sequences')
+        course_key = CourseKey.from_string(course_id)
         user = get_user_model().objects.get(username=(username or request.user.username))
-        serialized_onboarding_exam = ProctoredExamSerializer(onboarding_exam).data
+        details = learning_sequences_service.get_user_course_outline_details(
+            course_key, user, pytz.utc.localize(datetime.max)
+        )
+
+        for onboarding_exam in onboarding_exams:
+            usage_key = BlockUsageLocator.from_string(onboarding_exam.content_id)
+            if usage_key not in details.outline.accessible_sequences:
+                onboarding_exams.remove(onboarding_exam)
 
-        if not user.has_perm('edx_proctoring.can_take_proctored_exam', serialized_onboarding_exam):
+        if not onboarding_exams:
```

Comment:
Can we do a log.info in this case to inform splunk with the user_id who cannot see any onboarding exams in which course?